utils/directories.cc

```diff
@@ -77,21 +77,20 @@ directories::directories(bool developer_mode)
 { }
 
 future<> directories::create_and_verify(directories::set dir_set) {
-    return do_with(std::vector<file_lock>(), [this, dir_set = std::move(dir_set)] (std::vector<file_lock>& locks) {
-        return parallel_for_each(dir_set.get_paths(), [this, &locks] (fs::path path) {
-            return touch_and_lock(path).then([path = std::move(path), developer_mode = _developer_mode, &locks] (file_lock lock) {
-                locks.emplace_back(std::move(lock));
-                return disk_sanity(path, developer_mode).then([path = std::move(path)] {
-                    return directories::verify_owner_and_mode(path).handle_exception([](auto ep) {
-                        startlog.error("Failed owner and mode verification: {}", ep);
-                        return make_exception_future<>(ep);
-                    });
-                });
-            });
-        }).then([this, &locks] {
-            std::move(locks.begin(), locks.end(), std::back_inserter(_locks));
-        });
+    std::vector<file_lock> locks;
+    co_await parallel_for_each(dir_set.get_paths(), [this, &locks] (fs::path path) -> future<> {
```

Comment:
nit: please use coroutine::parallel_for_each. It is slightly more efficient for coroutines.